ComponentKit/Core/Scope/CKComponentScopeHandle.h

```diff
@@ -51,6 +51,11 @@
  return nil until `resolve` is called.
  */
 @property (nonatomic, strong, readonly) CKComponentController *controller;
+@property (nonatomic, weak, readonly) CKComponent *acquiredComponent;
+@property (nonatomic, assign, readonly) BOOL resolved;
+
+@property (nonatomic, assign, readonly) Class componentClass;
```

Comment:
exposing all these individual pieces of data publicly makes me feel pretty bad — but I don't have a good enough grasp to suggest an alternative. let's land it, and i can always try to poke around and see what i could change later…